client_test.go

```diff
@@ -39,6 +39,14 @@ func TestNewClient(t *testing.T) {
 	assert.Equal(t, conf.PollingWaitTime, defaultPollingWaitTime)
 }
 
+func TestEmptyEndpoint(t *testing.T) {
```

Comment:
Let's call this `TestInvalidConfig` because that's what we actually care about when verifying the client.

(Specific valid and invalid URLs are covered by `cluster_test.go`.)